### PR TITLE
[VIRTS-1951] Make Operation IDs UUID

### PR DESCRIPTION
**Pull request by b17zr**
## Description

- Change id generation to UUID
- Remove casting as int in Python and JavaScript

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Operation creation
- Operation status change
- Operation deletion
- Manual / potential link generation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I ~have made~ will make corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/objects/c_operation.py
Comment (bworrell) [inline app/objects/c_operation.py:22]: There is a `UUID` field type in marsmallow. Do we want to use that?

Comment (b17zr) [reply app/objects/c_operation.py:22]: We'd have to then parse ids as UUID when searching, right?

Comment (bworrell) [reply app/objects/c_operation.py:22]: What searching?

Comment (b17zr) [reply app/objects/c_operation.py:22]: `data_svc.locate()` or directly searching through the `ram`

File: app/objects/c_operation.py
Comment (bworrell) [inline app/objects/c_operation.py:59]: In the `__init__` method itself, should we cast `self.id = str(id)`? I'm wondering if there would be any issues if someone passed in an integer.

**Review by bworrell — CHANGES_REQUESTED** [review]
Just left one question. Otherwise I pulled this down along with the debrief and gameboard branches, ran an operation and clicked through those plugins and everything seemed to work as expected.

**Review by bworrell — COMMENTED** [review]
Were we going to bump the commit hashes on the gameboard/debrief submodules now (after they are merged) or hold off until the next release of caldera?

**Comment by b17zr** [comment]
> Were we going to bump the commit hashes on the gameboard/debrief submodules now (after they are merged) or hold off until the next release of caldera?

That's the plan. Figured we'd have to get https://github.com/mitre/gameboard/pull/28 and https://github.com/mitre/debrief/pull/41 merged first, then either update this PR or create a new one? Is there a preference there?

**Comment by bworrell** [comment]
> > Were we going to bump the commit hashes on the gameboard/debrief submodules now (after they are merged) or hold off until the next release of caldera?
> 
> That's the plan. Figured we'd have to get [mitre/gameboard#28](https://github.com/mitre/gameboard/pull/28) and [mitre/debrief#41](https://github.com/mitre/debrief/pull/41) merged first, then either update this PR or create a new one? Is there a preference there?

Hmmmm, let's include it in this PR.

**Review by bworrell — APPROVED** [review]
LGTM! Thanks for bumping the submodules!

**Review by wbooth — APPROVED** [review]
good

**Pull request merged**
